onnx/onnx.proto

```diff
@@ -237,6 +245,172 @@ message TensorProto {
 message SparseTensorProto {
   // The dimensions in the tensor.
   repeated int64 dims = 1;
+  // This field MUST be present this version of the IR.
   optional TensorProto indices = 2;
+  // This field MUST be present this version of the IR.  
   optional TensorProto values = 3;
 }
+
+// A ValueProto represents a value that may be serialized into a model.
+message ValueProto {
+
+  message KeyValuePairProto {
+    oneof key {
+      string s = 1;
+      int32 i32 = 2;
+      int64 i64 = 3;
+      uint64 ui64 = 4;
+    };
+    optional ValueProto value = 100;
+  }
+
+  message NameValuePairProto {
+    optional string key = 1;
+    optional ValueProto value = 100;
+  }
+
+  // Defines a record in its serialized format.
+  // A record is a sequence of one or more distinctly
```

Comment:
Why "distinctly"?

Reply:
I presume it indicates that the field value types are heterogenous, unlike a map where value types are homogeneous. Is that what 'distinctly typed' refers to?

Reply:
Yes. Heterogeneous.  If someone has better language, party on.